### PR TITLE
feat: add implementation for flownode instance entity, filter, sort, transformer

### PR DESCRIPTION
**Pull request by ralfpuchert**
## Description

* Add implementation for entity, filter, sort, transformer
* Test PR will follow
* Is rebased on [operate/c8-flownode-instances-endpoint](https://github.com/camunda/camunda/tree/operate/c8-flownode-instances-endpoint) 

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: service/src/main/java/io/camunda/service/entities/FlownodeInstanceEntity.java [hunk 1]
```diff
@@ -7,4 +7,20 @@
  */
 package io.camunda.service.entities;
 
-public record FlownodeInstanceEntity() {}
```
Comment (houssain-barouni) [inline service/src/main/java/io/camunda/service/entities/FlownodeInstanceEntity.java:10]: The term “FlowNode” is a compound word, and using proper camel case would improve readability.
I suggest using`FlowNodeInstanceEntity` which is consistent with existing naming in Operate
Same in the other classes

File: service/src/main/java/io/camunda/service/search/filter/FlownodeInstanceFilter.java [hunk 2]
```diff
@@ -7,15 +7,151 @@
...
+import static io.camunda.util.CollectionUtil.collectValuesAsList;
+
 import io.camunda.util.ObjectBuilder;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
 
-public record FlownodeInstanceFilter() implements FilterBase {
+public record FlownodeInstanceFilter(
+    List<Long> keys,
```
Comment (houssain-barouni) [inline service/src/main/java/io/camunda/service/search/filter/FlownodeInstanceFilter.java:19]: I wonder if we should use `flowNodeKeys` to be consistent with the other filters
We have used `decisionRequirementsKeys` for instance to be consistent with Zeebe API namings, but it seems that FlowNodeInstance is not part of the Zeebe API

**Review by houssain-barouni — COMMENTED** [review]
Looks good 👍 , I have 2 comments about namings

File: service/src/main/java/io/camunda/service/entities/FlownodeInstanceEntity.java (hunk 1 above)
Comment (nathansandi) [inline service/src/main/java/io/camunda/service/entities/FlownodeInstanceEntity.java:10]: I agree with @houssain-barouni ove renaming this class to `FlowNodeInstanceEntity`.

**Review by nathansandi — COMMENTED** [review]
Great work here,
I would just check the comments by @houssain-barouni

Also related to the key, in order to check if this entity also already exists over zeebe records to follow the same naming standard

**Review by ghost — APPROVED** [review]

**Comment by mihail-ca** [comment]
@ralfpuchert could you please make sure that **all** PRs are linked to the [corresponding issue](https://github.com/camunda/camunda/issues/20657), for easier tracking ? Same for [incidents endpoint issue](https://github.com/camunda/camunda/issues/20661), thank you
image

File: service/src/main/java/io/camunda/service/search/filter/FlownodeInstanceFilter.java (hunk 2 above)
Comment (ralfpuchert) [reply service/src/main/java/io/camunda/service/search/filter/FlownodeInstanceFilter.java:19]: Good point, it would then `flowNodeInstanceKeys` etc.

**Pull request merged**